src/widget/woverview.h

```diff
@@ -130,20 +101,41 @@ class WOverview : public WWidget, public TrackDropTarget {
 
     void updateCues(const QList<CuePointer> &loadedCues);
 
+    inline int length() {
+        return m_orientation == Qt::Horizontal ? width() : height();
+    }
+
+    inline int breadth() {
+        return m_orientation == Qt::Horizontal ? height() : width();
+    }
+
+    ConstWaveformPointer getWaveform() const {
+        return m_pWaveform;
+    }
+
+    double getTrackSamples() const {
+        if (m_trackLoaded) {
+            return m_trackSamplesControl->get();
+        } else {
+            // Ignore the value, because the engine can still have the old track
+            // during loading
+            return 0.0;
+        }
+    }
+
+    // Hold the last visual sample processed to generate the pixmap
+
     const QString m_group;
     UserSettingsPointer m_pConfig;
-    parented_ptr<ControlProxy> m_endOfTrackControl;
+
+    int m_type;
```

Comment:
Sure, we can that's why I mentioned it as nice-to-have in the description, and I already took a stab at it.
But I gave up since it took to long and isn't mandatory IMO, even feels like this makes the preferences more complicated (or I simply messed something up there)

I can take another look, but actually I'd appreciate if we could this merged soonish and refine later on (before the 2.5 release)